### ISSUE TITLE
Daily started to fail with msbuild JsonToTypedJsonCsMsBuildTaskNoLocking task

### ISSUE THREAD

**Issue by alemoi** [Starcounter/Starcounter#3054]
Here is the stack trace of fail:

```
error MSB4018: The "JsonToTypedJsonCsMsBuildTaskNoLocking" task failed unexpectedly. [C:\BuildAgent\work\sc-develop-daily-379\Level1\src\Server\Starcounter.Server.Rest\Starcounter.Server.Rest.csproj]
System.IO.FileNotFoundException: Could not load file or assembly 'Starcounter.XSON.JsTemplateParser, Version=0.2.1.0, Culture=neutral, PublicKeyToken=d2df1e81d0ca3abf' or one of its dependencies. The system cannot find the file specified. [C:\BuildAgent\work\sc-develop-daily-379\Level1\src\Server\Starcounter.Server.Rest\Starcounter.Server.Rest.csproj]
File name: 'Starcounter.XSON.JsTemplateParser, Version=0.2.1.0, Culture=neutral, PublicKeyToken=d2df1e81d0ca3abf' [C:\BuildAgent\work\sc-develop-daily-379\Level1\src\Server\Starcounter.Server.Rest\Starcounter.Server.Rest.csproj]
 [C:\BuildAgent\work\sc-develop-daily-379\Level1\src\Server\Starcounter.Server.Rest\Starcounter.Server.Rest.csproj]
Server stack trace:  [C:\BuildAgent\work\sc-develop-daily-379\Level1\src\Server\Starcounter.Server.Rest\Starcounter.Server.Rest.csproj]
   at Starcounter.Internal.MsBuild.JsonToCsMsBuildTask.ExecuteTask(ITaskItem[] InputFiles, ITaskItem[] OutputFiles, TaskLoggingHelper msbuildLog) [C:\BuildAgent\work\sc-develop-daily-379\Level1\src\Server\Starcounter.Server.Rest\Starcounter.Server.Rest.csproj]
   at Starcounter.Internal.MsBuild.JsonToTypedJsonCsMsBuildTaskNoLocking.Execute() [C:\BuildAgent\work\sc-develop-daily-379\Level1\src\Server\Starcounter.Server.Rest\Starcounter.Server.Rest.csproj]
   at System.Runtime.Remoting.Messaging.StackBuilderSink._PrivateProcessMessage(IntPtr md, Object[] args, Object server, Object[]& outArgs) [C:\BuildAgent\work\sc-develop-daily-379\Level1\src\Server\Starcounter.Server.Rest\Starcounter.Server.Rest.csproj]
   at System.Runtime.Remoting.Messaging.StackBuilderSink.SyncProcessMessage(IMessage msg) [C:\BuildAgent\work\sc-develop-daily-379\Level1\src\Server\Starcounter.Server.Rest\Starcounter.Server.Rest.csproj]
 [C:\BuildAgent\work\sc-develop-daily-379\Level1\src\Server\Starcounter.Server.Rest\Starcounter.Server.Rest.csproj]
Exception rethrown at [0]:  [C:\BuildAgent\work\sc-develop-daily-379\Level1\src\Server\Starcounter.Server.Rest\Starcounter.Server.Rest.csproj]
   at System.Runtime.Remoting.Proxies.RealProxy.HandleReturnMessage(IMessage reqMsg, IMessage retMsg) [C:\BuildAgent\work\sc-develop-daily-379\Level1\src\Server\Starcounter.Server.Rest\Starcounter.Server.Rest.csproj]
   at System.Runtime.Remoting.Proxies.RealProxy.PrivateInvoke(MessageData& msgData, Int32 type) [C:\BuildAgent\work\sc-develop-daily-379\Level1\src\Server\Starcounter.Server.Rest\Starcounter.Server.Rest.csproj]
   at Microsoft.Build.Framework.ITask.Execute() [C:\BuildAgent\work\sc-develop-daily-379\Level1\src\Server\Starcounter.Server.Rest\Starcounter.Server.Rest.csproj]
   at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() [C:\BuildAgent\work\sc-develop-daily-379\Level1\src\Server\Starcounter.Server.Rest\Starcounter.Server.Rest.csproj]
   at Microsoft.Build.BackEnd.TaskBuilder.<ExecuteInstantiatedTask>d__26.MoveNext() [C:\BuildAgent\work\sc-develop-daily-379\Level1\src\Server\Starcounter.Server.Rest\Starcounter.Server.Rest.csproj]
 [C:\BuildAgent\work\sc-develop-daily-379\Level1\src\Server\Starcounter.Server.Rest\Starcounter.Server.Rest.csproj]
WRN: Assembly binding logging is turned OFF. [C:\BuildAgent\work\sc-develop-daily-379\Level1\src\Server\Starcounter.Server.Rest\Starcounter.Server.Rest.csproj]
To enable assembly bind failure logging, set the registry value [HKLM\Software\Microsoft\Fusion!EnableLog] (DWORD) to 1. [C:\BuildAgent\work\sc-develop-daily-379\Level1\src\Server\Starcounter.Server.Rest\Starcounter.Server.Rest.csproj]
Note: There is some performance penalty associated with assembly bind failure logging. [C:\BuildAgent\work\sc-develop-daily-379\Level1\src\Server\Starcounter.Server.Rest\Starcounter.Server.Rest.csproj]
To turn this feature off, remove the registry value [HKLM\Software\Microsoft\Fusion!EnableLog]. [C:\BuildAgent\work\sc-develop-daily-379\Level1\src\Server\Starcounter.Server.Rest\Starcounter.Server.Rest.csproj]
 [C:\BuildAgent\work\sc-develop-daily-379\Level1\src\Server\Starcounter.Server.Rest\Starcounter.Server.Rest.csproj]
```
so the reason is `Could not load file or assembly 'Starcounter.XSON.JsTemplateParser`

Maybe its some build order issues?

Here is an example of failed build: 
https://scbuildserver/viewLog.html?buildId=31743&buildTypeId=Starcounter_DevelopDaily&tab=buildLog#_focus=3138

**Comment by alemoi** [comment]
And its on one build server only, so must be some build order issue.

**Comment by alemoi** [comment]
Again some problems with Msbuild failing to resolve dependencies correctly. Possible fix pushed via fb097b54cf512e5e3acba7258655d076ee0b1f7f
